tests/cnf/ran/powermanagement/tests/cpufreq.go

```diff
@@ -0,0 +1,81 @@
+	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/powermanagement/internal/helper"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/powermanagement/internal/tsparams"
+	performancev2 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/performanceprofile/v2"
+	"k8s.io/utils/cpuset"
+)
+
+var _ = Describe("CPU frequency tuning tests change the core frequencies of isolated and reserved cores",
+	Label(tsparams.LabelCPUFrequency), func() {
+		var (
+			perfProfile             *nto.Builder
+			desiredReservedCoreFreq = performancev2.CPUfrequency(2500001)
+			desiredIsolatedCoreFreq = performancev2.CPUfrequency(2200001)
+			originalIsolatedCPUFreq performancev2.CPUfrequency
+			originalReservedCPUFreq performancev2.CPUfrequency
+		)
+
+		BeforeEach(func() {
+			var err error
+
+			perfProfile, err = helper.GetPerformanceProfileWithCPUSet()
+			Expect(err).ToNot(HaveOccurred(), "Failed to get performance profile")
+
+			By("getting isolated core ID")
+			isolatedCPUSet, err := cpuset.Parse(string(*perfProfile.Object.Spec.CPU.Isolated))
+			Expect(err).ToNot(HaveOccurred(), "Failed to get isolated cpu set")
+			isolatedCPUsList := isolatedCPUSet.List()
+			isolatedCPUNumber := isolatedCPUsList[0]
+
+			By("getting reserved core ID")
+			reservedCPUSet, err := cpuset.Parse(string(*perfProfile.Object.Spec.CPU.Reserved))
+			Expect(err).ToNot(HaveOccurred(), "Failed to get reserved cpu set")
+			reservedCPUsList := reservedCPUSet.List()
+			reservedCPUNumber := reservedCPUsList[0]
+
+			By("getting original isolated core frequency")
+			spokeCommand := fmt.Sprintf("cat /sys/devices/system/cpu/cpufreq/policy%v/scaling_max_freq |cat -",
+				isolatedCPUNumber)
+			cmdOut, err := cluster.ExecCommandOnSNO(raninittools.Spoke1APIClient, 3, spokeCommand)
+			Expect(err).ToNot(HaveOccurred(), "Failed to %s, error:%s", spokeCommand, cmdOut)
+			freqAsInt, err := strconv.Atoi(strings.TrimSpace(cmdOut))
+			Expect(err).ToNot(HaveOccurred(), "strconv.Atoi Failed")
+			originalIsolatedCPUFreq = performancev2.CPUfrequency(freqAsInt)
+
+			By("getting original reserved core frequency")
+			spokeCommand = fmt.Sprintf("cat /sys/devices/system/cpu/cpufreq/policy%v/scaling_max_freq |cat -",
+				reservedCPUNumber)
+			cmdOut, err = cluster.ExecCommandOnSNO(raninittools.Spoke1APIClient, 3, spokeCommand)
+			Expect(err).ToNot(HaveOccurred(), "Failed to %s, error: %s", spokeCommand, cmdOut)
+			freqAsInt, err = strconv.Atoi(strings.TrimSpace(cmdOut))
+			Expect(err).ToNot(HaveOccurred(), "strconv.Atoi Failed")
+			originalReservedCPUFreq = performancev2.CPUfrequency(freqAsInt)
+
+		})
+
+		AfterEach(func() {
+			By("reverting the CPU frequencies to the original setting")
+			err := helper.SetCPUFreq(perfProfile, &originalIsolatedCPUFreq, &originalReservedCPUFreq)
+			Expect(err).ToNot(HaveOccurred(), "Failed to set CPU Freq")
+		})
+
+		When("reserved and isolated core frequency is configured via PerformanceProfile", func() {
+
```

Comment:
nitpick, but this blank line can be removed

Reply:
Removed- thanks.